### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by kaustavb12**
## Description

Initial implementation of Video iframe XBlock

## Testing instructions

1. Install this xblock in devstack
2. Go to advanced settings in any course and add `video_iframe` in the advanced module list
3. Open a new unit and click on `Advanced` -> `Video` (Warning: there might be block named `Video` under Advanced)
4.  Click on `Edit` and add the following URLs:
`Video URL` : `https://www.youtube.com/embed/c6YBGhZBB80?si=VM44_mtr_3hl6EOq`,
`Video Download URL`: `https://asu-moshe-test.s3.amazonaws.com/edX+%E2%80%93%C2%A0How+It+Works.mp4`,
`Captions Download URL`: `https://asu-moshe-test.s3.amazonaws.com/External+Video-en.srt`

## Other information

Opencraft Internal Ticket : [BB-8792](https://tasks.opencraft.com/browse/BB-8792)

### CODE REVIEW

File: video_iframe/public/js/src/video_iframe.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    wrapper_width = $('.video_wrapper', element).width();
+    $('.video_wrapper', element).height(wrapper_width*0.6);
+
+    video_download_link = data['video_download_link'];
+    captions_download_link = data['captions_download_link'];
+    display_name = data['display_name']
+    iframe_link = data['iframe_link']
+
+    if(video_download_link || video_download_link!==""){
```
Comment (CefBoud) [inline video_iframe/public/js/src/video_iframe.js:27]: Can you please explain why do we need ` video_download_link!==""`?

**Review by CefBoud — APPROVED** [review]
@kaustavb12 Great job on this!
I left two minor comments but other than that, LGTM :heavy_check_mark:.

- [X] I tested this in Tutor
- [X] I read through the code

File: video_iframe/public/js/src/video_iframe.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    var handlerUrl = runtime.handlerUrl(element, 'increment_count');
+
+    $('p', element).click(function(eventObject) {
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify({"hello": "world"}),
+            success: updateCount
+        });
+    });
```
[the remark this answers is not in the document]
Comment (kaustavb12) [reply video_iframe/public/js/src/video_iframe.js]: Ahhh... sorry about that. Good catch :+1:

File: video_iframe/public/js/src/video_iframe.js (hunk 1 above)
Comment (kaustavb12) [reply video_iframe/public/js/src/video_iframe.js]: For abundance of caution I guess, no other reason to justify this. Removed. :P

**Pull request merged**